tests/bases/test_composition.py

```diff
@@ -557,3 +553,19 @@ def test_compositional_metrics_update():
 
     assert compos.metric_a._num_updates == 3
     assert compos.metric_b._num_updates == 3
+
+
+def test_compositional_metrics_forward():
+
+    compos = DummyMetric(5) + DummyMetric(4)
```

Comment:
lets try to test composition with metric and something else
```suggestion
@pytest.mark.parametrize("metric_b", [4, DummyMetric(4)])
def test_compositional_metrics_forward(metric_b):
    """ test forward method of compositional metric """
    compos = DummyMetric(5) + metric_b
```